llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -1947,6 +1947,9 @@ void BaseMemOpClusterMutation::collectMemOpRecords(
     LocationSize Width = 0;
     if (TII->getMemOperandsWithOffsetWidth(MI, BaseOps, Offset,
                                            OffsetIsScalable, Width, TRI)) {
+      if (!Width.hasValue())
```

Comment:
I'd somewhat expect this to just collect the unknown width and let the users figure out what to do with it? Would that require widespread changes?

Reply:
Yeah, I think we need to modify `clusterNeighboringMemOps` and other places as well. That can be a future work if you agree.